test/e2e/schedulingprofile/intree.go

```diff
@@ -0,0 +1,196 @@
+limitations under the License.
+*/
+
+package schedulingprofile
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	fedcorev1a1 "github.com/kubewharf/kubeadmiral/pkg/apis/core/v1alpha1"
+	"github.com/kubewharf/kubeadmiral/pkg/controllers/scheduler"
+	"github.com/kubewharf/kubeadmiral/pkg/controllers/util"
+	"github.com/kubewharf/kubeadmiral/test/e2e/framework"
+	"github.com/kubewharf/kubeadmiral/test/e2e/framework/policies"
+	"github.com/kubewharf/kubeadmiral/test/e2e/framework/resources"
+)
+
+var scheduleTimeout = time.Second * 10
+
+var _ = ginkgo.Describe("Scheduling Profile", func() {
+	f := framework.NewFramework("scheduling-profile", framework.FrameworkOptions{CreateNamespace: true})
+
+	testPlacementFilter := func(ctx context.Context, profile *fedcorev1a1.SchedulingProfile, enabled bool) {
+		clusters, err := f.HostFedClient().CoreV1alpha1().FederatedClusters().List(ctx, metav1.ListOptions{})
+		gomega.Expect(err).ToNot(gomega.HaveOccurred(), framework.MessageUnexpectedError)
+		if len(clusters.Items) == 0 {
+			ginkgo.Fail("Test requires at least one federated cluster")
+		}
+
+		ginkgo.By("Creating scheduling profile")
+		profile, err = f.HostFedClient().CoreV1alpha1().SchedulingProfiles().Create(ctx, profile, metav1.CreateOptions{})
+		gomega.Expect(err).ToNot(gomega.HaveOccurred(), framework.MessageUnexpectedError)
+		ginkgo.DeferCleanup(func(ctx ginkgo.SpecContext) {
+			if err := f.HostFedClient().CoreV1alpha1().SchedulingProfiles().Delete(ctx, profile.Name, metav1.DeleteOptions{}); err != nil &&
+				!apierrors.IsNotFound(err) {
+				ginkgo.GinkgoLogr.Error(
+					err,
+					"failed to cleanup scheduling profile",
+					"profile-name",
+					profile.Name,
+				)
+			}
+		})
+
+		ginkgo.By("Creating propagation policy that references scheduling profile")
+		policy := policies.PropagationPolicyForClustersWithPlacements(f.Name(), []*fedcorev1a1.FederatedCluster{
+			{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: clusters.Items[0].Name,
+				},
+			},
+		})
+		policy.Spec.SchedulingProfile = profile.Name
+		policy, err = f.HostFedClient().
+			CoreV1alpha1().
+			PropagationPolicies(f.TestNamespace().Name).
+			Create(ctx, policy, metav1.CreateOptions{})
+		gomega.Expect(err).ToNot(gomega.HaveOccurred(), framework.MessageUnexpectedError)
+
+		ginkgo.By("Creating configmap that references propagation policy")
+		configMap := resources.GetSimpleConfigMap(f.Name())
+		policies.SetPropagationPolicy(configMap, policy)
+		configMap, err = f.HostKubeClient().CoreV1().ConfigMaps(f.TestNamespace().Name).Create(ctx, configMap, metav1.CreateOptions{})
+		gomega.Expect(err).ToNot(gomega.HaveOccurred(), framework.MessageUnexpectedError)
+
+		ginkgo.By("Verifying scheduling result")
+		if enabled {
```

Comment:
nit: this if branch can be nested in the eventually since the logic for getting placement is shared